docs/agent-sdk-architecture.md

```
@@ -216,6 +216,136 @@ if (result.stuck) {
+
+// Control execution
+await conversation.pause();
+await conversation.resume();
+
+// Confirmation flow
+await conversation.approveAction();
+await conversation.rejectAction('Too risky');
+```
+
+**Internal Agent Loop**:
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)